From a3ac8d7e3e44f59b01349978a7c71e9fbf1d764d Mon Sep 17 00:00:00 2001
From: Graham Moore <grmoore@altera.com>
Date: Wed, 6 Nov 2013 15:53:49 -0600
Subject: [PATCH 152/248] FogBugz #166244: Compilation error in spi-dw-pl330.c

Updated calls to device_prep_slave_sg() because the parameter list
had changed since the last time spi-dw-pl330.c was compiled.

[Original patch taken from
git://git.rocketboards.org/linux-socfpga.git socfpga-3.14]

Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
Signed-off-by: Graham Moore <grmoore@altera.com>
---
 drivers/spi/spi-dw-pl330.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/spi/spi-dw-pl330.c b/drivers/spi/spi-dw-pl330.c
index 050b9b4..413a382 100644
--- a/drivers/spi/spi-dw-pl330.c
+++ b/drivers/spi/spi-dw-pl330.c
@@ -21,6 +21,7 @@
 
 #include <linux/dma-mapping.h>
 #include <linux/dmaengine.h>
+#include <linux/amba/bus.h>
 #include <linux/amba/pl330.h>
 #include <linux/interrupt.h>
 #include <linux/slab.h>
@@ -165,7 +166,8 @@ static int spi_pl330_dma_transfer(struct dw_spi *dws, int cs_change)
 				&dws->tx_sgl,
 				1,
 				DMA_MEM_TO_DEV,
-				DMA_PREP_INTERRUPT);
+				DMA_PREP_INTERRUPT,
+				NULL);
 	txdesc->callback = spi_pl330_dma_done;
 	txdesc->callback_param = dws;
 
@@ -195,7 +197,8 @@ static int spi_pl330_dma_transfer(struct dw_spi *dws, int cs_change)
 				&dws->rx_sgl,
 				1,
 				DMA_DEV_TO_MEM,
-				DMA_PREP_INTERRUPT);
+				DMA_PREP_INTERRUPT,
+				NULL);
 	rxdesc->callback = spi_pl330_dma_done;
 	rxdesc->callback_param = dws;
 
-- 
1.9.1

